apps/src/lab2/views/dialogs/GenericPrompt.tsx

```diff
@@ -19,45 +21,66 @@ export type GenericPromptProps = Required<Pick<GenericDialogProps, 'title'>> & {
 type GenericPromptBodyProps = {
   placeholder?: string;
   prompt: string;
-  setPrompt: (newPrompt: string) => void;
+  handlePromptChange: (newPrompt: string) => void;
+  errorMessage?: string;
 };
 
 const GenericPromptBody: React.FunctionComponent<GenericPromptBodyProps> = ({
   placeholder,
   prompt,
-  setPrompt,
+  handlePromptChange,
+  errorMessage,
 }) => {
   return (
     <TextField
       name="prompt-field"
       placeholder={placeholder}
       value={prompt}
-      onChange={e => setPrompt(e.target.value)}
+      onChange={e => handlePromptChange(e.target.value)}
+      errorMessage={errorMessage}
     />
   );
 };
 
 const GenericPrompt: React.FunctionComponent<GenericPromptProps> = ({
   title,
   handleConfirm,
+  handleCancel,
   placeholder,
+  validateInput = () => undefined,
```

Comment:
This is a node-style error reporting function. Use it to validate input in the prompt in whatever arbitrary manner is useful.

Should return undefined if the input is valid, and otherwise return a string to display with the error message for how the user can correct their input error.